ivy/utils/exceptions.py

```diff
@@ -3,27 +3,41 @@
 from typing import Callable
 import sys
 import traceback as tb
+import io
 
+global buffer
 
 # Helpers #
 # ------- #
 
 
-def _log_stack_trace_truncated(trace_mode, func_wrapper_trace_mode):
+def _log_stack_trace_truncated(trace_mode, func_wrapper_trace_mode, buffer=None):
     if trace_mode in ["frontend", "ivy"]:
-        print(
-            "<stack trace is truncated to {} specific files,".format(trace_mode),
-            "call `ivy.set_exception_trace_mode('full')` to view the full trace>",
-        )
+        if buffer is not None:
```

Comment:
For readability and easier debugging, let's just write to the buffer and not print at all in all the helpers, and we'd then only do the printing inside the `_custom_exception_handle` and just rename all helpers having `_print` to `_write`